src/diracx/routers/job_manager/__init__.py

```diff
@@ -233,14 +236,101 @@ def __init__(self, user_info: UserInfo, allInfo: bool = True):
 
 
 @router.delete("/")
-async def delete_bulk_jobs(job_ids: Annotated[list[int], Query()]):
+async def delete_bulk_jobs(
+    job_ids: Annotated[list[int], Query()],
+    config: Annotated[Config, Depends(ConfigSource.create)],
+    job_db: JobDB,
+    job_logging_db: JobLoggingDB,
+    task_queue_db: TaskQueueDB,
+    background_task: BackgroundTasks,
+):
+    # TODO: implement job policy
+    try:
+        await asyncio.gather(
+            *(
+                delete_job(
+                    job_id,
+                    config,
+                    job_db,
+                    job_logging_db,
+                    task_queue_db,
+                    background_task,
+                )
+                for job_id in job_ids
+            )
+        )
+    except NoResultFound as e:
```

Comment:
We should probably use a [`TaskGroup`](https://docs.python.org/3/library/asyncio-task.html#asyncio.TaskGroup)+`except*` instead of `gather` so we get the better exception handling.

Ideally I think we should return a more explicit error like:

```python
detail={
    "message": "Failed to remove 5 jobs out of 167",
    "failed_ids": [1, 78, 32, 25, 90]
}
````

or maybe even:

```python
detail={
    "message": "Failed to remove 5 jobs out of 167",
    "failed_ids": {
        "1": str(e1),
        "78": str(e2),
        "32": str(e3),
        "25": str(e4),
        "90": str(e5)
    }
}
````

Reply:
Do you think we should cancel any action if one of them fail ? (for example here should we remove the other 162 jobs or not ?)

Reply:
If that’s practical I think that would be nice. What do you think @fstagni @chaen?

Reply:
> If that’s practical I think that would be nice.

I concur.

Reply:
@chaen Agrees as well

Reply:
Linked discussion: #97